packages/auto_mappr/lib/src/extensions/executable_element_extension.dart

```diff
@@ -1,5 +1,12 @@
 import 'package:analyzer/dart/element/element.dart';
 
 extension ExecutableElementExtension on ExecutableElement {
-  String get referCallString => isStatic ? '${enclosingElement.displayName}.$displayName' : displayName;
+  String get referCallString => isStatic &&
+          // A top-level function is static but the enclosing element has no
+          // display name. This could result in the expression being
+          // `.someFunction()` which is invalid. Therefore, we only add the
+          // enclosing element's display name if it is not empty.
```

Comment:
Nice catch, we probably use only static methods from Mappr so we didn't catch this. Would be adding a test for this also a good idea?

Nit: could you format the text so it is one sentence per line and if the sentence is too long you break it to meaningful parts? Wanted to post a link to a guide I try to follow but damn I cannot find it. 😅

Reply:
Ha, https://sembr.org !

Reply:
Added some tests for whenSourceIsNull which explicitly test static / top-level functions and static values.

Reply:
We already have `default_target` and `default_field` tests, so I would instead (then creating another test that tests basically the same) just make sure that both of these files have at least one top-level function aka moving one function from Mappr class to a top-level.

Reply:
That would cover the edge case but the test wouldn't be there for it so refactoring down the line might remove the test case since it doesn't exit explicitly for that use case. That's just my (very minor) concern with going that way